### ISSUE TITLE
Automatically catch the source of NaNs in objects

### ISSUE THREAD

**Issue by ma-sadeghi** [PMEAL/OpenPNM#1433]
Consider the following code snippet:
```
import numpy as np
import openpnm as op


net = op.network.Cubic(shape=[10, 10])
air = op.phases.Air(network=net)

Ps1 = np.random.choice(net.Ps, size=net.Np//5, replace=False)
Ts1 = net.find_neighbor_throats(Ps1)
geom1 = op.geometry.StickAndBall(network=net, pores=Ps1, throats=Ts1)

Ps2 = np.setdiff1d(net.Ps, Ps1)
Ts2 = np.setdiff1d(net.Ts, Ts1)
geom2 = op.geometry.StickAndBall(network=net, pores=Ps2, throats=Ts2)

phys1 = op.physics.Standard(network=net, phase=air, geometry=geom1)
phys2 = op.physics.Standard(network=net, phase=air, geometry=geom2)

fd = op.algorithms.FickianDiffusion(network=net, phase=air)
fd.set_value_BC(pores=net.pores("left"), values=1.0)
fd.set_value_BC(pores=net.pores("right"), values=0.3)
fd.run()
```
Since `throat.endpoints`, which is a `geom1` model, requires `pore.diameter` to be known in the entire domain, it'll be NaNs in some places, and so the code fails to converge, leaving the user clueless about where the problem comes from. If the user is OpenPNM-savvy, probably they'll figure out to put `geom1.regenerate_models()` after `geom2` instantiation, and everything will be good to go. However, we should find a way of at least informing the user where the problem comes from and possibly how to fix it via a warning.

**Comment by jgostick** [comment]
Maybe a more general/simple fix should be for the solver to catch nans, and report their presence, with an additional comment like 'try regenerating models on all objects'.

Also, speaking of regenerating models on all objects, the project could/should have a method to do this, which scans all objects in the correct order and does it.

**Comment by jgostick** [comment]
My comment above pertains to #1398

**Comment by ma-sadeghi** [comment]
@jgostick I like what you proposed. Since the order of regeneration also matters, this can be done via the project object

**Comment by ma-sadeghi** [comment]
This is now fixed in PR #1458
